Buoy_tool/views.py

```diff
@@ -56,14 +60,23 @@ def buoy(request, buoy_id):
     for key in metaDic:
         idList.append(key['id'])
     if buoy_id in idList:
+        url_nc = "http://tds.glos.us/thredds/dodsC/buoy_agg_standard/{0}/{0}.ncml".format(buoy_id)
+        try:
+            ds = Dataset(url_nc)
```

Comment:
Is this going to make a request to NCML every time a user visits a buoy page?

Reply:
This hits the aggregation endpoint, but doesn't fetch any variable data.  It only goes by the `time_coverage_end` global attribute.  AFAIK this aggregation is based upon time anyhow and I haven't encountered long load times when requesting the aggregations, nor have I encountered high memory usage on the client side due to the lazily-loaded nature of OPeNDAP datasets in Python NetCDF4.  That said, if there is a more preferable way to do this, I'm open to hearing it.

Reply:
I dont know anything about this app but I do know that there is a JSON file that gets routinely updated with the data:
https://glbuoys.glos.us/static/Buoy_tool/data/meta_english.json
I would suggest fetching that attribute once during that process and putting it in that JSON file rather than fetching it in a view. There are potentially thousands of daily users of this site.

Reply:
I'll see what can be done.  Ideally I'd like to move away from JSON files to something like a database in the future. There's already an "updated" field in some of the JSON entries, but that appears to be pulling from GLCFS, rather than THREDDS.  It's also not in each deployment.  I've sent you the folder with the routines for generating this JSON data.  It's not all entirely clear to me how it's operating at the moment, but I will try to see if it can be added rather than making a request to THREDDS directly.